src/infrastructure/validation/testing/TestHarness.java

```diff
@@ -0,0 +1,395 @@
+/**
+ * @author        Abhishek Saran
+ * @author        Manas Sanjay
+ * module         Infrastructure module 
+ * team           Test Harness
+ * description    Test Harness main class
+ * summary        To provide functionality to tester to run test case sets in various ways, 
+ *                Methods such as runAll(), runByName(), runByCategory() and runByPriority() are provided. 
+ */
+package infrastructure.validation.testing;
+import java.io.File;
+import java.util.*;
+import infrastructure.validation.logger.LoggerFactory;
+import infrastructure.validation.logger.ILogger;
+import infrastructure.validation.logger.LogLevel;
+import infrastructure.validation.logger.ModuleID;
+
+public class TestHarness{
+  
+  private String savePath;
+  
+  public void setSavePath(String path){
+    this.savePath = path;
+  }
+  
+  public String getSavePath(){
+    return this.savePath;
+  }
+  
+  /**
+  * static helper method to return all the test cases of project
+  * @return ArrayList of File type containing all the test cases with valid naming
+  */
+  private static ArrayList<File> getAllTests(){
+    ILogger logger = LoggerFactory.getLoggerInstance();
+    ArrayList<File> allTests = new ArrayList<File>();
+    try{
+      File[] modules = new File("src/").listFiles(File::isDirectory); 
```

Comment:
Assumes that the CWD (current working directory) is the base of the repo KK.
Should be fine @convict-git ?

Reply:
we have used relative path earlier which works fine while executing  in the terminal but doesn't work well in eclipse project. This will not work well in the terminal but will work fine in eclipse project. I am not sure why this is happening.

Reply:
Reference - https://stackoverflow.com/questions/2792870/java-cant-find-file-when-running-through-eclipse

Reply:
When you create a default Java application in Eclipse, you get this directory structure:

./ProjectName/ - root directory
./ProjectName/bin/ - output directory, containing .class files
./ProjectName/src/ - source directory, containing .java files

If your application requests "./data.txt" it will search for it relative to the root directory.

Reply:
Agree. You're searching for `src` in the base of the repo.